File: sasmodels/models/sasfit_langevin_static_up_up.c
///////////////////////////////////////////////////
//    This is automatically genearted file       //
//    by sasfit_convert.py                       //
//    Some editting might be required            //
///////////////////////////////////////////////////

double Iq( double q, double NU,  double B0,  double B1,  double NU,
           double RHO_0,  double PHASE_PAR,  double PSI_DEG,  double SQ_AMPL,
           double SQ_PHASE,  double SQ_STATIC,  double FNUC_FMAG,  double MU_KT,
           double LAM_CENTER,  double D_LAM_LAM,  double SD,  double P_POL,
           double EPSILON_FLIPPER,  double T_PLUS,  double T_MINUS,  double B_STATIC,
           double B_STAT_PHI_DEG,  double DELTA_T_BOTTOM,  double DELTA_T_TOP,
           double ALPHA_PHASE,  double LAMBDA,  double MU,  double DAMP_ALPHA,
           double SQ_TIME,  double TEMPERATURE,  double AD,  double AD_FUNC,
           double NONAD_FUNC,  double FNUC,  double FMAG,  double S1,  double S2,
           double R,  double R_AV,  double TIME,  double STATIC,  double PSI);
double Fq( double q,  double NU,  double B0,  double B1,  double NU,
           double RHO_0,  double PHASE_PAR,  double PSI_DEG,  double SQ_AMPL,
           double SQ_PHASE,  double SQ_STATIC,  double FNUC_FMAG,  double MU_KT,
           double LAM_CENTER,  double D_LAM_LAM,  double SD,  double P_POL,
           double EPSILON_FLIPPER,  double T_PLUS,  double T_MINUS,  double B_STATIC,
           double B_STAT_PHI_DEG,  double DELTA_T_BOTTOM,  double DELTA_T_TOP,
           double ALPHA_PHASE,  double LAMBDA,  double MU,  double DAMP_ALPHA,
           double SQ_TIME,  double TEMPERATURE,  double AD,  double AD_FUNC,
           double NONAD_FUNC,  double FNUC,  double FMAG,  double S1,  double S2,
           double R,  double R_AV,  double TIME,  double STATIC,  double PSI);
double form_volume(  double NU,  double B0,  double B1,  double NU,
                     double RHO_0,  double PHASE_PAR,  double PSI_DEG,  double SQ_AMPL,
                     double SQ_PHASE,  double SQ_STATIC,  double FNUC_FMAG,  double MU_KT,
                     double LAM_CENTER,  double D_LAM_LAM,  double SD,  double P_POL,
                     double EPSILON_FLIPPER,  double T_PLUS,  double T_MINUS,  double B_STATIC,
                     double B_STAT_PHI_DEG,  double DELTA_T_BOTTOM,  double DELTA_T_TOP,
                     double ALPHA_PHASE,  double LAMBDA,  double MU,  double DAMP_ALPHA,
                     double SQ_TIME,  double TEMPERATURE,  double AD,  double AD_FUNC,
                     double NONAD_FUNC,  double FNUC,  double FMAG,  double S1,  double S2,
                     double R,  double R_AV,  double TIME,  double STATIC,  double PSI);
double Iqxy( double qx, double qy, double NU, double B0, double B1, double NU,
             double RHO_0, double PHASE_PAR, double PSI_DEG, double SQ_AMPL, double SQ_PHASE,
             double SQ_STATIC, double FNUC_FMAG, double MU_KT, double LAM_CENTER,
             double D_LAM_LAM, double SD, double P_POL, double EPSILON_FLIPPER,
             double T_PLUS, double T_MINUS, double B_STATIC, double B_STAT_PHI_DEG,
             double DELTA_T_BOTTOM, double DELTA_T_TOP, double ALPHA_PHASE, double LAMBDA,
             double MU, double DAMP_ALPHA, double SQ_TIME, double TEMPERATURE, double AD,
             double AD_FUNC, double NONAD_FUNC, double FNUC, double FMAG, double S1,
             double S2, double R, double R_AV, double TIME, double STATIC, double PSI);
/*
* Author(s) of this file:
*   Joachim Kohlbrecher (joachim.kohlbrecher@psi.ch)
*/
double Iq( double q, double NU,  double B0,  double B1,  double NU,
           double RHO_0,  double PHASE_PAR,  double PSI_DEG,  double SQ_AMPL,
           double SQ_PHASE,  double SQ_STATIC,  double FNUC_FMAG,  double MU_KT,
           double LAM_CENTER,  double D_LAM_LAM,  double SD,  double P_POL,
           double EPSILON_FLIPPER,  double T_PLUS,  double T_MINUS,  double B_STATIC,
           double B_STAT_PHI_DEG,  double DELTA_T_BOTTOM,  double DELTA_T_TOP,
           double ALPHA_PHASE,  double LAMBDA,  double MU,  double DAMP_ALPHA,
           double SQ_TIME,  double TEMPERATURE,  double AD,  double AD_FUNC,
           double NONAD_FUNC,  double FNUC,  double FMAG,  double S1,  double S2,
           double R,  double R_AV,  double TIME,  double STATIC,  double PSI)
{
    PSI = PSI_DEG*M_PI/180.0;
    AD = 1.0;
    R=1.0;
    R_AV=R;
    TEMPERATURE = 273.0;
    DAMP_ALPHA = 1.0;
    STATIC = 1.0;
    P_POL = -fabs(P_POL);
    EPSILON_FLIPPER = 1.0;
    T_PLUS = 1.0;
    T_MINUS = 0.0;
    return i_measured(t,NU, B0, B1, NU, RHO_0, PHASE_PAR, PSI_DEG, SQ_AMPL,
                      SQ_PHASE, SQ_STATIC, FNUC_FMAG, MU_KT, LAM_CENTER, D_LAM_LAM, SD, P_POL,
                      EPSILON_FLIPPER, T_PLUS, T_MINUS, B_STATIC, B_STAT_PHI_DEG, DELTA_T_BOTTOM,
                      DELTA_T_TOP, ALPHA_PHASE, LAMBDA, MU, DAMP_ALPHA, SQ_TIME, TEMPERATURE, AD,
                      AD_FUNC, NONAD_FUNC, FNUC, FMAG, S1, S2, R, R_AV, TIME, STATIC, PSI);
}
double Fq( double q,  double NU,  double B0,  double B1,  double NU,
           double RHO_0,  double PHASE_PAR,  double PSI_DEG,  double SQ_AMPL,
           double SQ_PHASE,  double SQ_STATIC,  double FNUC_FMAG,  double MU_KT,
           double LAM_CENTER,  double D_LAM_LAM,  double SD,  double P_POL,
           double EPSILON_FLIPPER,  double T_PLUS,  double T_MINUS,  double B_STATIC,
           double B_STAT_PHI_DEG,  double DELTA_T_BOTTOM,  double DELTA_T_TOP,
           double ALPHA_PHASE,  double LAMBDA,  double MU,  double DAMP_ALPHA,
           double SQ_TIME,  double TEMPERATURE,  double AD,  double AD_FUNC,
           double NONAD_FUNC,  double FNUC,  double FMAG,  double S1,  double S2,
           double R,  double R_AV,  double TIME,  double STATIC,  double PSI)
{
// insert your code here
    return 0.0;
}
double form_volume(  double NU,  double B0,  double B1,  double NU,
                     double RHO_0,  double PHASE_PAR,  double PSI_DEG,  double SQ_AMPL,
                     double SQ_PHASE,  double SQ_STATIC,  double FNUC_FMAG,  double MU_KT,
                     double LAM_CENTER,  double D_LAM_LAM,  double SD,  double P_POL,
                     double EPSILON_FLIPPER,  double T_PLUS,  double T_MINUS,  double B_STATIC,
                     double B_STAT_PHI_DEG,  double DELTA_T_BOTTOM,  double DELTA_T_TOP,
                     double ALPHA_PHASE,  double LAMBDA,  double MU,  double DAMP_ALPHA,
                     double SQ_TIME,  double TEMPERATURE,  double AD,  double AD_FUNC,
                     double NONAD_FUNC,  double FNUC,  double FMAG,  double S1,  double S2,
                     double R,  double R_AV,  double TIME,  double STATIC,  double PSI)
{
// insert your code here
    return 0.0;
}
double Iqxy( double qx, double qy, double NU, double B0, double B1, double NU,
             double RHO_0, double PHASE_PAR, double PSI_DEG, double SQ_AMPL, double SQ_PHASE,
             double SQ_STATIC, double FNUC_FMAG, double MU_KT, double LAM_CENTER,
             double D_LAM_LAM, double SD, double P_POL, double EPSILON_FLIPPER,
             double T_PLUS, double T_MINUS, double B_STATIC, double B_STAT_PHI_DEG,
             double DELTA_T_BOTTOM, double DELTA_T_TOP, double ALPHA_PHASE, double LAMBDA,
             double MU, double DAMP_ALPHA, double SQ_TIME, double TEMPERATURE, double AD,
             double AD_FUNC, double NONAD_FUNC, double FNUC, double FMAG, double S1,
             double S2, double R, double R_AV, double TIME, double STATIC, double PSI)
{
    double q = sqrt(qx*qx + qy*qy);
    return Iq( q, NU, B0, B1, NU, RHO_0, PHASE_PAR, PSI_DEG, SQ_AMPL, SQ_PHASE,
               SQ_STATIC, FNUC_FMAG, MU_KT, LAM_CENTER, D_LAM_LAM, SD, P_POL, EPSILON_FLIPPER,
               T_PLUS, T_MINUS, B_STATIC, B_STAT_PHI_DEG, DELTA_T_BOTTOM, DELTA_T_TOP,
               ALPHA_PHASE, LAMBDA, MU, DAMP_ALPHA, SQ_TIME, TEMPERATURE, AD, AD_FUNC,
               NONAD_FUNC, FNUC, FMAG, S1, S2, R, R_AV, TIME, STATIC, PSI);
}

File: sasmodels/models/sasfit_langevin_static_up_up.py
r"""
This file has been automatically gereated by sasfit_convert
The model calculates an empirical functional form for SAS data characterized
by langevin_static_up_up

Definition:
-----------

References:
-----------

"""
from numpy import inf

name = "langevin_static_up_up"
title = " "
description = ""
category = " "
#pylint: disable=bad-whitespace, line-too-long
parameters = [
 [ "NU", 	"", 	10.0, 	[-inf, inf], 	"", 	""],
 [ "B0", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "B1", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "NU", 	"", 	1.0, 	[-inf, inf], 	"", 	""],
 [ "RHO_0", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "PHASE_PAR", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "PSI_DEG", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "SQ_AMPL", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "SQ_PHASE", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "SQ_STATIC", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "FNUC_FMAG", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "MU_KT", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "LAM_CENTER", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "D_LAM_LAM", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "SD", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "P_POL", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "EPSILON_FLIPPER", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "T_PLUS", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "T_MINUS", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "B_STATIC", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "B_STAT_PHI_DEG", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "DELTA_T_BOTTOM", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "DELTA_T_TOP", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "ALPHA_PHASE", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "LAMBDA", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "MU", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "DAMP_ALPHA", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "SQ_TIME", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "TEMPERATURE", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "AD", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "AD_FUNC", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "NONAD_FUNC", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "FNUC", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "FMAG", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "S1", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "S2", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "R", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "R_AV", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "TIME", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "STATIC", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "PSI", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
]
 #pylint: enable=bad-whitespace, line-too-long

source = [  "sasfit_langevin_static_up_up.c" ]

demo = dict(
	NU = 10.0,
	B0 = 0.0,
	B1 = 0.0,
	NU = 1.0,
	RHO_0 = 0.0,
	PHASE_PAR = 0.0,
	PSI_DEG = 0.0,
	SQ_AMPL = 0.0,
	SQ_PHASE = 0.0,
	SQ_STATIC = 0.0,
	FNUC_FMAG = 0.0,
	MU_KT = 0.0,
	LAM_CENTER = 0.0,
	D_LAM_LAM = 0.0,
	SD = 0.0,
	P_POL = 0.0,
	EPSILON_FLIPPER = 0.0,
	T_PLUS = 0.0,
	T_MINUS = 0.0,
	B_STATIC = 0.0,
	B_STAT_PHI_DEG = 0.0,
	DELTA_T_BOTTOM = 0.0,
	DELTA_T_TOP = 0.0,
	ALPHA_PHASE = 0.0,
	LAMBDA = 0.0,
	MU = 0.0,
	DAMP_ALPHA = 0.0,
	SQ_TIME = 0.0,
	TEMPERATURE = 0.0,
	AD = 0.0,
	AD_FUNC = 0.0,
	NONAD_FUNC = 0.0,
	FNUC = 0.0,
	FMAG = 0.0,
	S1 = 0.0,
	S2 = 0.0,
	R = 0.0,
	R_AV = 0.0,
	TIME = 0.0,
	STATIC = 0.0,
	PSI = 0.0)
